statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,325 @@
+                                       approx_hess_cs, approx_fprime_cs)
+
+class GenericTruncated(CountModel):
+    __doc__ = """
+    Generic Truncated model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, truncation=0,offset=None,
+                       exposure=None, missing='none', **kwargs):
+        super(GenericTruncated, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+        self.exog = self.exog[self.endog >= (truncation + 1)]
+        self.endog = self.endog[self.endog >= (truncation + 1)]
+        self.trunc = truncation
+
+    def loglike(self, params):
+        """
+        Loglikelihood of Generic Truncated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+
+        Notes
+        --------
+        .. math:: \\ln L=\\sum_{y_{i}=0}\\ln(w_{i}+(1-w_{i})*P_{main\\_model})+
+            \\sum_{y_{i}>0}(\\ln(1-w_{i})+L_{main\\_model})
+            where P - pdf of main model, L - loglike function of main model.
+
+        """
+        return np.sum(self.loglikeobs(params))
+
+    def loglikeobs(self, params):
+        """
+        Loglikelihood for observations of Generic Truncated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : ndarray (nobs,)
+            The log likelihood for each observation of the model evaluated
+            at `params`. See Notes
+
+        Notes
+        --------
+        .. math:: \\ln L=\\ln(w_{i}+(1-w_{i})*P_{main\\_model})+
+            \\ln(1-w_{i})+L_{main\\_model}
+            where P - pdf of main model, L - loglike function of main model.
+
+        for observations :math:`i=1,...,n`
+
+        """
+        llf_main = self.model_main.loglikeobs(params)
+
+        pmf = np.zeros_like(self.endog)
+        for i in range(self.trunc + 1):
+            model = self.model_main_name(np.ones_like(self.endog) * i,
+                                         self.exog)
+            pmf +=  np.exp(model.loglikeobs(params))
```

Comment:
this loop could be put into a `cdf` method that can be overwritten by a subclass.
E.g. poisson should have a build in method (or method delegated to scipy distribution cdf)